File: EngSoft.Jogo.Patos/src/engsoft/jogo/patos/Pato_Bravo.java
package engsoft.jogo.patos;

// Classe Concreta
// Herança
// Polimorfismo
public class Pato_Bravo extends Pato implements Padrao_Grasnar {

    public Pato_Bravo() {
        setComportamento(new Voaveis_Asa());
    }

    public String mostrar() {
        return "Eu sou o Pato Bravo.";
    }

    public String grasnar() {
        return "Que-Que. Grrrrrrrrr.";
    }

}

File: EngSoftTypeScript/src/PatoBravo.ts
import {Pato} from "./Pato";
import {PadraoGrasnar} from "./PadraoGrasnar";
import {VoaveisAsa} from "./VoaveisAsa";

export class PatoBravo extends Pato implements PadraoGrasnar {

    constructor() {
        super();
        this.comportamentoVoar = new VoaveisAsa();
    }

    public mostrar(): string {
        return "Eu sou o Pato Bravo.";
    }

    public grasnar(): string {
        return "Que-Que. Grrrrrrrrr.";
    }

}